### PR TITLE
lfs/batcher: introduce test cases

### PR DESCRIPTION
**Pull request by ttaylorr**
This commit adds the notion of a test case into the `lfs.Batcher` tests. Each
test case has three parameters: a batch size, the number of items to add to that
batcher, and whether or not it should exit the batcher after adding all items.

This commit also adds a helper func called `assertAll` which processes a set of
test cases, throwing assertion errors into the `*testing.T` along the way.

On the other hand, this commit does not contain any utilities to assert that a
deadlock will occur when feeding items into a Batcher, i.e., when the batch size
is set to 0.

### CODE REVIEW

File: lfs/batcher_test.go [hunk 1]
```diff
@@ -7,28 +7,66 @@ import (
...
+	assertAll([]batcherTestCase{
+		{2, 4, true},
+		{3, 5, true},
+		{0, 0, true},
+	}, t)
+}
+
+// Type batcherTestCase specifies information about how to run a particular test
+// around the type lfs.Batcher.
+type batcherTestCase struct {
```
Comment (technoweenie) [inline lfs/batcher_test.go]: Go comments for types and funcs start with the name. The only godoc comments that have a prefix are package comments.  https://golang.org/doc/effective_go.html#commentary

File: lfs/batcher_test.go [hunk 2]
```diff
@@ -7,28 +7,66 @@ import (
...
+		return 0
 	}
 
-	b.Exit()
+	return b.ItemCount / b.BatchSize
+}
+
+// Func assertAll processes all test cases, throwing assertion errors if they
+// fail.
+func assertAll(cases []batcherTestCase, t *testing.T) {
```
Comment (technoweenie) [inline lfs/batcher_test.go]: I'd like to see a more descriptive name here, like `runBatcherTests`. A name like `assertAll` seems too generic here. This seems to be more than just a simple assertion, as well.

**Comment by ttaylorr** [comment]
Thanks for the suggestions, @technoweenie.  I've pushed commits to address both of your comments.

**Comment by technoweenie** [comment]
Thanks for the tests!

File: lfs/batcher_test.go [hunk 3]
```diff
@@ -7,28 +7,66 @@ import (
...
+// around the type lfs.Batcher.
+type batcherTestCase struct {
+	BatchSize  int
+	ItemCount  int
+	ShouldExit bool
+}
 
-	for i := 0; i < 4; i++ {
-		b.Add(&Downloadable{})
+// Batcher makes and retunrs a lfs.Batcher according to the specification given
```
Comment (WillHipschman) [inline lfs/batcher_test.go]: nit: typo, should be "returns"

**Comment by ttaylorr** [comment]
Thanks for the review @technoweenie, @WillHipschman!

File: lfs/batcher_test.go (hunk 3 above)
Comment (technoweenie) [inline lfs/batcher_test.go]: Doh! 3f1eec26a25b0c8e17f8b89c3d3dd419189d80c5

Comment (ttaylorr) [inline lfs/batcher_test.go]: Thanks for fixing that!  Totally missed it.

**Pull request merged**